src/code/z_schedule.c

```diff
@@ -0,0 +1,282 @@
+
+    result->result = cmd->result;
+    result->time0 = time0;
+    result->time1 = time1;
+    result->hasResult = true;
+
+    return true;
+}
+
+s32 Schedule_CheckBeforeTimeS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckBeforeTimeS* cmd = (ScheduleCmdCheckBeforeTimeS*)*script;
+    f32 f;
+    u16 testTime;
+    u16 now;
+
+    SCHEDULE_CALC_TIME(cmd->timeHr, cmd->timeMin, testTime, f);
+
+    now = SCHEDULE_TIME_NOW;
+
+    if (now < testTime) {
+        *script = *script + cmd->offset;
+    }
+
+    return false;
+}
+
+s32 Schedule_CheckBeforeTimeL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdCheckBeforeTimeL* cmd = (ScheduleCmdCheckBeforeTimeL*)*script;
+    f32 f;
+    u16 testTime;
+    u16 now;
+
+    SCHEDULE_CALC_TIME(cmd->timeHr, cmd->timeMin, testTime, f);
+
+    now = SCHEDULE_TIME_NOW;
+
+    if (now < testTime) {
+        *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+    }
+
+    return false;
+}
+
+s32 Schedule_BranchS(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdBranchS* cmd = (ScheduleCmdBranchS*)*script;
+
+    *script = *script + cmd->offset;
+    return false;
+}
+
+s32 Schedule_BranchL(GlobalContext* globalCtx, u8** script, ScheduleResult* result) {
+    ScheduleCmdBranchL* cmd = (ScheduleCmdBranchL*)*script;
+
+    *script = *script + (s16)((cmd->offsetH << 8) | cmd->offsetL);
+    return false;
+}
+
+static s32 (*sCmdFuncs[])(GlobalContext*, u8**, ScheduleResult*) = {
+    Schedule_CheckFlagS,       Schedule_CheckFlagL,   Schedule_CheckTimeRangeS, Schedule_CheckTimeRangeL,
+    Schedule_ReturnValueL,     Schedule_ReturnNone,   Schedule_ReturnEmpty,     Schedule_Nop,
+    Schedule_CheckMiscS,       Schedule_ReturnValueS, Schedule_CheckSceneS,     Schedule_CheckSceneL,
+    Schedule_CheckDayS,        Schedule_CheckDayL,    Schedule_ReturnTime,      Schedule_CheckBeforeTimeS,
+    Schedule_CheckBeforeTimeL, Schedule_BranchS,      Schedule_BranchL,
+};
+
+static u8 sCmdSizes[] = {
+    sizeof(ScheduleCmdCheckFlagS),
+    sizeof(ScheduleCmdCheckFlagL),
+    sizeof(ScheduleCmdCheckTimeRangeS),
+    sizeof(ScheduleCmdCheckTimeRangeL),
+    sizeof(ScheduleCmdReturnValueL),
+    sizeof(ScheduleCmdBase),
+    sizeof(ScheduleCmdBase),
+    sizeof(ScheduleCmdNop),
+    sizeof(ScheduleCmdCheckMiscS),
+    sizeof(ScheduleCmdReturnValueS),
+    sizeof(ScheduleCmdCheckSceneS),
+    sizeof(ScheduleCmdCheckSceneL),
+    sizeof(ScheduleCmdCheckDayS),
+    sizeof(ScheduleCmdCheckDayL),
+    sizeof(ScheduleCmdReturnTime),
+    sizeof(ScheduleCmdCheckBeforeTimeS),
+    sizeof(ScheduleCmdCheckBeforeTimeL),
+    sizeof(ScheduleCmdBranchS),
+    sizeof(ScheduleCmdBranchL),
+};
+
+s32 Schedule_RunScript(GlobalContext* globalCtx, u8* script, ScheduleResult* result) {
+    u8 size;
+    s32 stop;
+
+    do {
+        size = sCmdSizes[*script];
+        stop = (*sCmdFuncs[*script])(globalCtx, &script, result);
```

Comment:
```suggestion
        size = sScheduleCmdSizes[*script];
        stop = (*sScheduleCmdFuncs[*script])(globalCtx, &script, result);
```
Even though these are local, I think it wouldn't hurt to have `Schedule` in the name